core/embed/trezorhal/usb.c

```diff
@@ -153,11 +153,19 @@ void usb_start(void) { USBD_Start(&usb_dev_handle); }
 void usb_stop(void) { USBD_Stop(&usb_dev_handle); }
 
 secbool usb_configured(void) {
-  USBD_HandleTypeDef *pdev = &usb_dev_handle;
+  const USBD_HandleTypeDef *pdev = &usb_dev_handle;
   if (pdev->dev_state == USBD_STATE_CONFIGURED) {
     return sectrue;
   }
 
+  // Linux has autosuspend device after 2 seconds by default.
+  // So a suspended device that was seen as configured is reported as
+  // configured.
+  if (pdev->dev_state == USBD_STATE_SUSPENDED &&
+      pdev->dev_old_state == USBD_STATE_CONFIGURED) {
```

Comment:
at what point will `dev_old_state` get pushed out?
i.e., can we get suspended and then suspended again and push out the "previously configured" state?

Reply:
You can't get "doubly suspended", you can enter from configured->suspended just once. Then again suspended->configured and back, but not suspended->suspended.

Here is the state machine:

![image](https://user-images.githubusercontent.com/812173/187476662-4f92c1b4-cacb-4b64-b05a-6f73f7498321.png)